Client/Frontend/Home/BookmarksEditPanel.swift

```diff
@@ -0,0 +1,335 @@
+        guard let source = self.source,
+              let bookmark = self.bookmark else {
+            return
+        }
+        
+        guard let factory = source.modelFactory.value.successValue else {
+            log.error("Couldn't get model factory. This is unexpected.")
+            self.onModelFailure(DatabaseError(description: "Unable to get factory."))
+            return
+        }
+        
+        if let err = factory.updateByGUID(BookmarkItem(guid: bookmark.guid, title: titleField?.text ?? bookmark.title, url: urlField?.text ?? bookmark.url)).value.failureValue {
+            log.debug("Failed to remove \(bookmark.guid).")
+            self.onModelFailure(err)
+            return
+        }
+        
+        NSNotificationCenter.defaultCenter().postNotificationName(BookmarkStatusChangedNotification, object: bookmark, userInfo:["added": false])
+        self.navigationController?.popViewControllerAnimated(true)
+    }
+    
+    override func tableView(tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        if section != 0 {
+            return nil
+        }
+        
+        // Don't show a header for the root
+        if source == nil || parentFolders.isEmpty {
+            return nil
+        }
+        guard let header = tableView.dequeueReusableHeaderFooterViewWithIdentifier(BookmarkFolderEditHeaderViewIdentifier) as? BookmarkFolderEditTableViewHeader else { return nil }
+        
+        doneButton = header.doneButton
+        
+        header.doneButton.removeTarget(self, action: #selector(BookmarksEditPanel.doneAndSave(_:)), forControlEvents: UIControlEvents.TouchUpInside)
+        header.doneButton.addTarget(self, action: #selector(BookmarksEditPanel.doneAndSave(_:)), forControlEvents: UIControlEvents.TouchUpInside)
+        
+        // register as delegate to ensure we get notified when the user interacts with this header
+        if header.delegate == nil {
+            header.delegate = self
+        }
+        
+        if parentFolders.count == 1 {
+            header.textLabel?.text = NSLocalizedString("Bookmarks", comment: "Panel accessibility label")
+        } else if let parentFolder = parentFolders.last {
+            header.textLabel?.text = parentFolder.title
+        }
+        header.contentView.backgroundColor = UIColor(red: 250/255, green: 250/255, blue: 250/255, alpha: 1)
+        
+        return header
+    }
+    func tableView(tableView: UITableView, willSelectRowAtIndexPath indexPath: NSIndexPath) -> NSIndexPath? {
+        if indexPath.row == 2 {
+            return indexPath
+        }
+        return nil
+    }
+    
+    func tableView(tableView: UITableView, shouldHighlightRowAtIndexPath indexPath: NSIndexPath) -> Bool {
+        return indexPath.row == 2;
+    }
+    
+    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        tableView.deselectRowAtIndexPath(indexPath, animated: false)
+        
```

Comment:
Actually we should only be doing this for the 'delete button'. I would replace the delete button with this call to invoke the deletion. Why make life difficult by adding a button in a cell when we can just tap the cell?